Makefile

```diff
@@ -318,7 +309,7 @@ ifeq ($(and $(FW_VER_LT_1_6),$(FW_VER_NE_0_00)),)
 	@ln -sf $(OUTPUT_IMAGE) $(TIANOCORE_ATF_IMAGE)
 else
 	$(eval TIANOCORE_ATF_SIGNED_IMAGE := $(WORKSPACE)/Build/$(BOARD_NAME_UFL)/$(BUILD_VARIANT)_$(EDK2_GCC_TAG)/$(BOARD_NAME)_tianocore_atf.img.signed)
-	$(eval DBU_KEY := $(EDK2_PLATFORMS_SRC_DIR)/Platform/Ampere/$(BOARD_NAME_UFL)Pkg/TestKeys/Dbu_AmpereTest.priv.pem)
+	$(eval DBU_KEY := $(EDK2_PLATFORMS_PKG_DIR)/TestKeys/Dbu_AmpereTest.priv.pem)
```

Comment:
Why did you change this location ? Ampere put keys to Platform/Ampere/JadePkg/TestKeys

Reply:
I did not change the location, EDK2_PLATFORMS_PKG_DIR equal to EDK2_PLATFORMS_SRC_DIR)/Platform/Ampere/$(BOARD_NAME_UFL)Pkg according to line 28.
Changing this expression makes me easier to replace the keys via override EDK2_PLATFORMS_PKG_DIR, but that would be the next modification to modify line 28 replacing ":=" with "?=".

Reply:
got it 

I think don't need to change ":=" with "?=" - it's defined with EDK2_PLATFORMS_SRC_DIR

Reply:
Because I cannot submit my platform code into "edk2-platforms" (=EDK2_PLATFORMS_SRC_DIR) due to company IP policy, so I make it a submodule of my project now, and the BOARD_NAME is different from "Jade", I have to replace them via changing EDK2_PLATFORMS_PKG_DIR. I think most OEM will encounter the similar situation because I have not seen any platform code other than "Jade".

Reply:
With ":=", you still pass it when you compile

`make EDK2_PLATFORMS_PKG_DIR=<your path> .....`

Reply:
you are right, I will cease the modification, thanks for the tip!